app.js

```diff
@@ -158,6 +158,36 @@ app.get('/bake', (request, response) => {
   response.send(getLayoutHTML(content));
 });
 
+// Page to greet the user based on a name passed to the url
+app.get('/greet', (request, response) => {
+  let content;
+  console.log(request.query);
+  if (!request.query.name) {
+    content = `
+    <h1>Whoops</h1>
+    <p>
+      I didn't get your name!
+    </p>
+    `;
+  } else {
+    let name = capitalize(request.query.name);
+    content = `
+    <h1>Greetings, ${name}</h1>
+    <p>
+      Thank you for stopping by this page!  You will see a lot of cool stuff with this repo.
+    </p>
+    `;
+  }
+  content += `
+    <form id="passName" method="GET" action="/greet">
+      <input type="text" id="name" name="name" placeholder="Your name" required>
```

Comment:
Make this change and see what happens:

```html
<input value="${name}" type="text" id="name" name="name" placeholder="Your name" required>
```

Reply:
It gives me an error

Reply:
Oh, uhh, right, it's only defined in the `else` block.

This:

```html
<input value="${request.query.name}" type="text" id="name" name="name" placeholder="Your name" required>
```